src/background/main.js

```diff
@@ -123,7 +123,7 @@ class Main {
 
     for (const observer of this.#observers) {
       try {
-        const result = await observer.handleEvent(type, data);
+        const result = observer.handleEvent(type, data);
```

Comment:
This is wrong, because here you collect promises and not the real result.

Reply:
Yes, my PR description discusses this – the use of the awaited queue here delays the `openPopup` action and triggers the bug "user gesture required" for me – the popup never opens.   

I've since restored the await and tried a different approach that fixes the popup bug (and hopefully doesn't break anything else!)